### PR TITLE
added chat functionality

### PR DESCRIPTION
**Pull request by B6Amine**
I managed to add the chat functionality that will make users be able to chat with others ( groomers and clients at the end ) . For now , the essential feature is working u can chat with a user by adding his phone number . It starts a discussion with this user but with the wrong user id . the UI still need to be improved . Unfortunately , the messages are appearing in reverse order and the discussion is consider autonome , the share of user id's is not perfect yet . I'll try to fix all that before the M3 deadline .

image
image
image

### CODE REVIEW

**Review by chtidaba — APPROVED** [review]
Good chat functionnality, implementation is globally good, Nice work

File: app/src/main/java/com/android/PetPamper/CAViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                    val user = userDocument.toObject<User>()
+                    userData.value = user
+                    inProgress.value = false
+                    populateChats()
+                } else {
+                    handleException(null, "User not found")
+                    inProgress.value = false
+                }
+            }
+    }
```
Comment (Shinsaegae) [inline app/src/main/java/com/android/PetPamper/CAViewModel.kt:88]: Handling user data should be done in the user data layer.

File: app/src/main/java/com/android/PetPamper/CAViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    fun depopulateChat() {
+        chatMessages.value = listOf()
+        currentChatMessagesListener = null
+    }
+
+
+}
```
Comment (Shinsaegae) [inline app/src/main/java/com/android/PetPamper/CAViewModel.kt:228]: Should ideally separate data logic and the UI logic, which the ViewModel should be for.

File: app/src/main/java/com/android/PetPamper/ui/chat/ChatListScreen.kt [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        Icon(
+            imageVector = Icons.Rounded.Add,
+            contentDescription = null,
+            tint = Color.White
+        )
+    }
+}
+
+
+
```
Comment (Shinsaegae) [inline app/src/main/java/com/android/PetPamper/ui/chat/ChatListScreen.kt:208]: The UI is well-implemented, and correctly separated from the UI logic.

File: app/src/main/java/com/android/PetPamper/ui/chat/SingleChatScreen.kt [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+            Icon(
+                imageVector = Icons.Rounded.Send,
+                contentDescription = "Send",
+                tint = Color(0xFF3F51B5)
+            )
+        }
+    }
+}
+
+
```
Comment (Shinsaegae) [inline app/src/main/java/com/android/PetPamper/ui/chat/SingleChatScreen.kt:191]: Same here, UI logic is correctly delegated to the ViewModel.

**Review by Shinsaegae — COMMENTED** [review]
The both the chat feature is implemented correctly (both the UI and the data layer).

However, the PR is quite large. Consider splitting the chat data handling and the UI into separate PRs in the future.

**Pull request merged**